### PR TITLE
macOS still needs --interactive and --tty set

### PR DESCRIPTION
**Pull request by raskchanky**
It turns out that on macOS, `--interactive` and `--tty` weren't getting passed when doing a `hab studio run` or `hab studio build`, which meant that coloring was no longer active and you couldn't ctrl+c to kill the process.  This fixes that breakage, and adds some new lines so the code is easier to read.

Fixes https://github.com/habitat-sh/habitat/issues/5661
Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by raskchanky** [comment]
Note: I've verified that this does the right thing on linux and macOS but have not yet verified Windows.

File: components/hab/src/command/studio/docker.rs [hunk 1]
```diff
@@ -226,47 +226,57 @@ fn run_container<I, J, S, T>(docker_cmd: PathBuf,
...
     let mut cmd_args: Vec<OsString> = vec!["run".into(), "--rm".into()];
+
     if !using_windows_containers {
         cmd_args.push("--privileged".into());
     }
+
     match args.first().map(|f| f.to_str().unwrap_or_default()) {
-        Some("build") | Some("run") => {}
+        Some("build") | Some("run") if cfg!(not(target_os = "macos")) => {
+        }
```
Comment (baumanj) [inline components/hab/src/command/studio/docker.rs]: So we need `--tty` and `--interactive` on *all* platforms if the command is neither `build` nor `run` (i.e., `enter`, `help`, `new`, `rm`, `version`)? But *only* Mac OS needs `--tty` and `--interactive` for `build` and `run` as well?

That seems…odd. Let's be sure we understand what's going on here, and if it's really that weird, let's make sure to give some explanation in a comment for the poor soul who comes across this next.

Comment (chefsalim) [reply components/hab/src/command/studio/docker.rs]: This seems the opposite of what is desired.. is the desire to NOT push those for Mac?
Docker studio builds break if these are not set correctly btw.

Comment (baumanj) [reply components/hab/src/command/studio/docker.rs]: From the description, the desire *is* to push `--tty` and `--interactive` in the Mac OS case. And this achieves that, but it's a bit confusing because its expressed as "if the `studio` command is `build` or `run` and the `target_os` is *NOT* `macos`, do nothing. Otherwise, push `--tty` and `--interactive`".

It might be easier to understand if reworked to express the positive action. That said, I'm still not clear as to why this behavior should be different just for `build` and `run`.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: To clarify:  this PR aims to fix https://github.com/habitat-sh/habitat/issues/5661.

The command in question is something like this:  `hab studio run 'hab pkg export docker core/redis'`

In Linux, running that command is fine - you get colored output and you can also ctrl+c out of it.  But, it's also not using docker in Linux, so that's probably a bit of a moot point.

In Windows, @mwrock tells me that you can't docker export from inside a studio, so that command isn't valid there.

It _is_ an issue on macOS, though, because on macOS, running that command does _not_ currently pass `--interactive` and `--tty` flags to docker, meaning that you can't ctrl+c to kill the process and you don't get colored output.  This fixes that problem.

To be honest, I have no idea why this behavior should be different just for `build` and `run`.

So in summary, Linux is fine, Windows is fine, macOS is broken, and this fix makes it work in macOS (and it continues to work in Linux).  I will add a comment explaining why this is the way it is, but I'm hesitant to change the logic any further without fully understanding why there are exceptions here for `build` and `run`.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: Ok, here's the history on this https://github.com/habitat-sh/habitat/commit/bfb3d818c47660ac5b7dac78b267c767ca07f4e1

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: Also https://github.com/habitat-sh/habitat/commit/ac153638ea57a59747bc44393ba20984e9b8d367

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: Further context https://github.com/habitat-sh/habitat/issues/5289

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: Also see this PR https://github.com/habitat-sh/habitat/pull/5469

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: So yeah, this entire thing feels fraught with peril.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: Also https://github.com/habitat-sh/habitat/pull/3336 which is where this match statement originated

Comment (baumanj) [reply components/hab/src/command/studio/docker.rs]: So, what if you do a a `hab studio -D run 'hab pkg export docker core/redis'` on linux? It seems like the issue is less about the platform and more about docker/chroot and interactive/CI.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: `hab studio -D run 'hab pkg export docker core/redis'` exhibits the broken behavior on current hab, 0.78.0, as well as with this branch, with linux.

Also note that this issue is less about `hab studio run` and `hab studio build` _per se_ and more about those 2 commands specifically when used in conjunction with a docker studio.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: But you're right.  This isn't as much about macOS, but more about docker inside the studio.

Comment (baumanj) [reply components/hab/src/command/studio/docker.rs]: Since this code is only germane to the docker studio, it sounds like the only reason we want to *omit* `--tty` and `--interactive` in when we're running in CI (or some other headless environment). So maybe instead of hardcoding this logic with platform assumptions, we should push `--tty` and `--interactive` by default, but support a `--no-tty` or `--non-interactive` flag that can be added for CI or similar use cases.

Comment (raskchanky) [reply components/hab/src/command/studio/docker.rs]: That sounds like the most sane solution to me.  I'll make the appropriate changes.  Thanks for talking through this.  I think the end result will be a lot better because of it.

**Comment by raskchanky** [comment]
I've verified that docker studio builds both on Linux and macOS are interactive by default, but become non-interactive when passing the `--no-tty` and `--non-interactive` flags.

**Comment by raskchanky** [comment]
@baumanj This is ready for a fresh look at your leisure.

**Comment by mwrock** [comment]
We will want to make sure that builder passes `--no-tty` and `--non-interactive` for its windows docker builds since the docker `tty` will distort some of the text.
